pom.xml

```diff
@@ -49,22 +49,29 @@
         <dependency>
             <groupId>fr.inria.gforge.spoon</groupId>
             <artifactId>spoon-core</artifactId>
-            <version>8.2.0-beta-12</version>
+            <version>8.3.0</version>
+            <exclusions>
+                <exclusion>
+                    <!-- must be excluded as it is signed, which causes problems with unsigned version from sonar -->
+                    <groupId>org.eclipse.jdt</groupId>
+                    <artifactId>org.eclipse.jdt.core</artifactId>
+                </exclusion>
+            </exclusions>
```

Comment:
@slarse Do you remember what problems were arising in the absence of its exclusion? I am using the local build of spoon with this [change](https://github.com/INRIA/spoon/pull/4320#issuecomment-988121027) in the comment, but sorald throws an error that it cannot find the `.get` method for `HashtableOfPackage`. I am assuming it is because of this line.

Reply:
As I recall, Sonar-java 6.9 pulled in (or bundled, maybe) an unsigned `jdt` package, while the one pulled in from Spoon is signed. Signed as in there's a GPG signature on it.

That caused errors on building it.

As both Spoon and Sonar-Java use jdt, you need to make sure that the version they depend on is the same. Otherwise you'll get strange errors.